Lib/test/test_tarfile.py

```diff
@@ -690,6 +691,32 @@ def format_mtime(mtime):
             tar.close()
             os_helper.rmtree(DIR)
 
+    @unittest.skipUnless(hasattr(errno, "EFTYPE"), "errno.EFTYPE required")
+    def test_extract_chmod_eftype(self):
+        # Extracting a file as non-root should skip the sticky bit (gh-108948)
+        # even on platforms where chmod fails with EFTYPE (i.e. FreeBSD). But
+        # we need to take care that any other error is preserved.
+        with ArchiveMaker() as arc:
+            arc.add("sticky", mode="?rw-rw-rwt")
+        tar = arc.open(errorlevel=2)
+        DIR = os.path.join(TEMPDIR, "chmod")
+        os.mkdir(DIR)
+        try:
+            # this should not raise:
+            tar.extract("sticky", DIR)
```

Comment:
Can you check permissions of the created file? Please add also an empty line for readability.

Reply:
I expect that if run as root, the test sets the sticky bit. Otherwise, it doesn't. Python has a Cirrus CI job which runs the Python test suite... as root!